ddl/partition.go

```diff
@@ -474,7 +474,7 @@ func getPartitionIDs(table *model.TableInfo) []int64 {
 func checkRangePartitioningKeysConstraints(sctx sessionctx.Context, s *ast.CreateTableStmt, tblInfo *model.TableInfo, constraints []*ast.Constraint) error {
 	// Returns directly if there is no constraint in the partition table.
 	// TODO: Remove the test 's.Partition.Expr == nil' when we support 'PARTITION BY RANGE COLUMNS'
-	if len(constraints) == 0 || s.Partition.Expr == nil {
+	if s.Partition.Expr == nil {
```

Comment:
Why remove "len(constraints) == 0"? 
Check table name for columns will be checked in `checkPartitionFuncType`.

Reply:
I want line 484 to be executed, `checkPartitionFuncType` will return error but the error code is not correct
`extractPartitionColumns` in line 484 will return the correct error for this bug fix.